src/export-data/export-data.service.ts

```diff
@@ -0,0 +1,101 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Workbook, Worksheet } from 'exceljs';
+import { createTransport } from 'nodemailer';
+
+@Injectable()
+export class ExportDataService {
+  constructor(private prisma: PrismaService) {}
+
+  public async exportGameData(
+    userId: string,
+    emailTo: string,
+  ): Promise<string> {
+    const user = await this.prisma.user
+      .findUniqueOrThrow({
+        where: {
+          role: 'ADMIN',
+          id: userId,
+        },
+      })
+      .catch(() => {
+        throw new ForbiddenException(
+          'Only user with role admin can export the game data',
+        );
+      });
+
+    const histories = await this.prisma.gameHistory.findMany({
+      orderBy: {
+        gameType: 'asc',
+      },
+    });
+
+    try {
+      const { worksheet, workbook, transporter } = this.setup();
+      worksheet.columns = [
+        { header: 'Game Type', key: 'gameType' },
+        { header: 'Score', key: 'score' },
+        { header: 'Start Time', key: 'startTime' },
+        { header: 'End Time', key: 'endTime' },
+      ];
+
+      histories.forEach((row) => {
+        worksheet.addRow({
+          gameType: row.gameType,
+          score: row.score,
+          startTime: new Date(row.startTime),
+          endTime: new Date(row.endTime),
+        });
+      });
+
+      const buffer = await workbook.xlsx.writeBuffer();
+      const currentDate = new Date();
+      const formattedDate = currentDate
+        .toLocaleDateString('en-GB', {
+          day: '2-digit',
+          month: '2-digit',
+          year: '2-digit',
+        })
+        .replace(/\//g, '-');
+
+      const mailOptions = {
+        from: process.env.EMAIL_SENDER,
+        to: emailTo,
+        subject: 'Game History Talacare',
+        text: 'Please find the exported game data attached',
+        attachments: [
+          {
+            filename: `game_history_${formattedDate}.xlsx`,
+            content: buffer,
+          },
+        ],
+      };
```

Comment:
untuk bagian penulisan data ke worksheet dan pembuatan email objectnya, sepertinya juga bisa dibuat jadi 1 method, supaya nanti kalau mau edit fungsionalitas terkait penulisan data-nya, bisa refer ke 1 tempat itu aja (?)

Reply:
hmm sbnrnya 1 1 function itu untuk nulis data nya kok to, trs gua mikirnya untuk yg date karena sbnrnya cuma tambahn function jd gpp di 1  funciton yg sama aja

Reply:
Oalah oke cell